File: Nian_Karl/QL/src/org/uva/ql/ast/expression/binary/Binary.java
package org.uva.ql.ast.expression.binary;

import org.uva.ql.ast.expression.Expression;

public abstract class Binary extends Expression {
	
	protected final Expression left;
	protected final Expression right;
	
	public Binary(Expression left, Expression right) {
		this.left = left;
		this.right = right;
	}
	
	public Expression getLeftExpression() {
		return left;
	}
	
	public Expression getRightExpression() {
		return right;
	}

}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Binaries/Binary.cs

namespace AST.Nodes.Expressions.Binaries
{
    public abstract class Binary : Expression
    {
        private readonly Expression left; 
        private readonly Expression right;

        protected Binary(Expression left, Expression right, PositionInText pos)
            :base(pos)
        {
            this.left = left; 
            this.right = right;
        }

        public Expression Left()
        { return left; }

        public Expression Right()
        { return right; }

    }
}
